### PR TITLE
feat: [PL-22408]: added scm userprofile validation to git-sync API

### PR DESCRIPTION
**Pull request by bhavya181**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)

### CODE REVIEW

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java [hunk 1]
```diff
@@ -117,4 +117,13 @@ private SourceCodeManagerDTO getScmUserProfile(String accountId, UserPrincipal u
     }
     return sourceCodeManagerDTO.get();
   }
+
+  public boolean validateIfScmUserProfileIsSet(String accountIdentifier) {
+    List<SourceCodeManagerDTO> sourceCodeManagerDTOS = sourceCodeManagerService.get(accountIdentifier);
+    if (sourceCodeManagerDTOS.isEmpty()) {
+      throw new InvalidRequestException("We don’t have your git credentials for the selected folder."
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java:124]: Selected folder ?

**Review by abhinav-harness — CHANGES_REQUESTED** [review]

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java, line 208 at r1](https://reviewable.io/reviews/harness/harness-core/30650#-MvUKp6cByutW2xCZwb2:-MvUKp6cByutW2xCZwb3:brbfm8q) ([raw file](https://github.com/harness/harness-core/blob/adac6c668677921f89628be9605a6c76fe76036b/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java#L208)):*
> ```Java
> 
>   private boolean isFullSyncEnabled(String accountIdentifier) {
>     return RestClientUtils.getResponse(
> ```

If this rest call fails because of a network issue, then our full sync will also fail. We can wrap this statement in a try-catch

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java, line 126 at r1](https://reviewable.io/reviews/harness/harness-core/30650#-MvULGH89568CvXXcT-O:-MvULGH89568CvXXcT-P:b24msbs) ([raw file](https://github.com/harness/harness-core/blob/adac6c668677921f89628be9605a6c76fe76036b/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java#L126)):*
> ```Java
>       throw new InvalidRequestException("We don’t have your git credentials for the selected folder."
>           + " Please update the credentials in user profile.");
>     }
> ```

What if the user profile is set, but it is for Bitbucket and the customer is using a github connector

**Review by DeepakPatankar — COMMENTED** [review]
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 2 change requests. 0 of 2 approvals obtained.

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java (hunk 1 above)
Comment (bhavya181) [reply 136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java:124]: Screenshot 2022-02-10 at 1 13 30 PM
This is what we show on Ui

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 2 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java, line 126 at r1](https://reviewable.io/reviews/harness/harness-core/30650#-MvULGH89568CvXXcT-O:-MvXequODHKu5O4Gu_YX:bqc507p) ([raw file](https://github.com/harness/harness-core/blob/adac6c668677921f89628be9605a6c76fe76036b/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java#L126)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

What if the user profile is set, but it is for Bitbucket and the customer is using a github connector
</blockquote></details>

We have to make those changes from UI side as well with bitbucket support. I will create a ticket for that.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java, line 208 at r1](https://reviewable.io/reviews/harness/harness-core/30650#-MvUKp6cByutW2xCZwb2:-MvXf2rX0dkT1b4SUjrU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/adac6c668677921f89628be9605a6c76fe76036b/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java#L208)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

If this rest call fails because of a network issue, then our full sync will also fail. We can wrap this statement in a try-catch
</blockquote></details>

Done.

**Comment by bhavya181** [comment]
trigger messagecheck

**Review by DeepakPatankar — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java, line 124 at r1](https://reviewable.io/reviews/harness/harness-core/30650#-MvU9_E0dze4eT1CGIoh-r1-124:-MvXjoEcDFU-dXsNTtK2:b-yktob6) ([raw file](https://github.com/harness/harness-core/blob/adac6c668677921f89628be9605a6c76fe76036b/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/UserProfileHelper.java#L124)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Screenshot 2022-02-10 at 1 13 30 PM
This is what we show on Ui
</blockquote></details>

same comment I have on other ticket :joy:

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 2 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java, line 212 at r2](https://reviewable.io/reviews/harness/harness-core/30650#-MvXk2aJ-zApq_Ix4Drz:-MvXk2aKBUSguOjob1iX:bbagbe9) ([raw file](https://github.com/harness/harness-core/blob/6c40a29ad376029369f72f5906720d500229a410/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java#L212)):*
> ```Java
>           accountClient.isFeatureFlagEnabled(FeatureName.NG_GIT_FULL_SYNC.name(), accountIdentifier));
>     } catch (Exception ex) {
>       log.error("Exception occurred while getting response", ex);
> ```

please improve the logline, it doesn't mention anything about the use-case

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650)* status: 3 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java, line 212 at r2](https://reviewable.io/reviews/harness/harness-core/30650#-MvXk2aJ-zApq_Ix4Drz:-MvXlb-B8RzVOwTMWmUl:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6c40a29ad376029369f72f5906720d500229a410/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/YamlGitConfigServiceImpl.java#L212)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

please improve the logline, it doesn't mention anything about the use-case
</blockquote></details>

Done.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 3 files at r1, 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30650#-:-MvcosL0-BMKyJCfxhmN:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**